python-avd/pyavd/_eos_designs/structured_config/base/__init__.py

```diff
@@ -163,8 +163,35 @@ def router_multicast(self) -> None:
 
     @structured_config_contributor
     def hardware_counters(self) -> None:
-        """hardware_counters set based on hardware_counters.features variable."""
-        self.structured_config.hardware_counters = self.inputs.hardware_counters
+        """
+        Set hardware_counters.
+
+        Contributing data sources:
+          - hardware_counters.features variable.
+          - platform_settings.feature_support.hardware_counters fact.
+        """
+        if not self.inputs.hardware_counters:
+            return
+        # Avoid collision with eos_cli_config_gen
+        if not self.shared_utils.platform_settings.feature_support.hardware_counters.supported:
+            self.custom_structured_configs.nested.hardware_counters = EosCliConfigGen.HardwareCounters._from_null()
+            return
+        hardware_counters = self.inputs.hardware_counters._cast_as(EosCliConfigGen.HardwareCounters)
```

Comment:
I don't think you need the cast as here, since the hardware_counters is reusing the eos cli config gen class. (Check if pyright complains when you remove the cast_as)